client/src/backoffice/InboxListPage.tsx

```diff
@@ -0,0 +1,262 @@
+import React, { FC, useState, useEffect, ChangeEvent, useRef } from 'react';
+
+/* Libraries */
+import ReactModal from 'react-modal';
+import ReactPaginate from 'react-paginate';
+
+/* Helpers */
+import { Notification, getNotifications, getEvents, PoapEvent } from '../api';
+
+/* Components */
+import { Loading } from '../components/Loading';
+
+/* Assets */
+import plus from '../images/plus.svg';
+
+const PAGE_SIZE = 10;
+
+type PaginateAction = {
+  selected: number;
+};
+
+const InboxListPage: FC = () => {
+  const [page, setPage] = useState<number>(0);
+  const [total, setTotal] = useState<number>(0);
+  const [shouldResetPage, setShouldResetPage] = useState<boolean>(false);
+  const [notificationType, setNotificationType] = useState<string>('inbox');
+  const [recipientFilter, setRecipientFilter] = useState<string>('everyone');
+  const [modalOpen, setModalOpen] = useState<boolean>(false);
+  const [selectedEvent, setSelectedEvent] = useState<number | undefined>(undefined);
+  const [modalText, setModalText] = useState<string>('');
+  const [modalTitle, setModalTitle] = useState<string>('');
+  const [isFetchingNotifications, setIsFetchingNotifications] = useState<null | boolean>(null);
+  const [notifications, setNotifications] = useState<null | Notification[]>(null);
+  const [events, setEvents] = useState<PoapEvent[]>([]);
+
+  useEffect(() => {
+    fetchEvents();
+  }, []);
+
+  useEffect(() => {
+    page !== 0 ? setPage(0) : fetchNotifications();
+  }, [notificationType]);
+
+  useEffect(() => {
+    setShouldResetPage(true);
+  }, [recipientFilter, selectedEvent]);
+
+  useEffect(() => {
+    if (shouldResetPage) {
+      setShouldResetPage(false);
+
+      if (recipientFilter === 'event') {
+        if (selectedEvent === undefined) return;
+      } else {
+        setSelectedEvent(undefined);
+      }
+
+      page !== 0 ? setPage(0) : fetchNotifications();
+    }
+  }, [shouldResetPage]);
+
+  useEffect(() => {
+    fetchNotifications();
+  }, [page]);
+
+  const fetchEvents = async () => {
+    const events = await getEvents();
+    setEvents(events);
+  };
+
+  const fetchNotifications = () => {
+    setIsFetchingNotifications(true);
+
+    let event_id = undefined;
+    if (recipientFilter === 'event' && selectedEvent !== undefined) {
+      event_id = selectedEvent > -1 ? selectedEvent : undefined;
+    }
+
+    getNotifications(PAGE_SIZE, page * PAGE_SIZE, notificationType, event_id)
+      .then(response => {
+        if (!response) return;
+        setNotifications(response.notifications);
+        setTotal(response.total);
+      })
+      .catch(error => console.error(error))
+      .finally(() => setIsFetchingNotifications(false));
+  };
+
+  const handlePageChange = (obj: PaginateAction) => {
+    setPage(obj.selected);
+  };
+
+  const handleRadio = (name: string, value: string) => {
+    if (name === 'notificationType') setNotificationType(value);
+    if (name === 'recipientFilter') setRecipientFilter(value);
+  };
```

Comment:
Can we type the `name` and `value` with those expected values?

You could use a [union type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#union-types). Please take a quick look at the docs.
Also if those types are used only in this file you can save them below your last import in this same file.
```typescript
type Name = 'notificationType' | 'recipientFilter';

type Value = 'inbox' | 'push' | 'everyone' | 'event';
```

```typescript
const handleRadio = (name: Name, value: Value) => {
  if (name === 'notificationType') setNotificationType(value);
  if (name === 'recipientFilter') setRecipientFilter(value);
};
```